### PR TITLE
chore: add state macvlan test

### PR DESCRIPTION
**Pull request by rollandf**
Add Sync tests

### CODE REVIEW

**Review by ykulazhenkov — APPROVED** [review]
thx, LGTM

**Comment by e0ne** [comment]
LGTM. @rollandf  could you please rebase this PR?

File: pkg/state/state_macvlan_network_test.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2024 NVIDIA
```
Comment (e0ne) [inline pkg/state/state_macvlan_network_test.go:2]: Please change to:
2024 NVIDIA CORPORATION & AFFILIATES

**Review by e0ne — CHANGES_REQUESTED** [review]

**Comment by rollandf** [comment]
Move to different package and use public API for the tests

**Review by vasrem — APPROVED** [review]
LGTM

**Review by ykulazhenkov — APPROVED** [review]

File: pkg/state/state_macvlan_network_test.go (hunk 1 above)
Comment (rollandf) [reply pkg/state/state_macvlan_network_test.go]: Done

**Review by e0ne — APPROVED** [review]

**Pull request merged**